### PR TITLE
Julia interface documentation

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: src/julia_interface.jl [hunk 1]
```diff
@@ -1,5 +1,19 @@
 export objcons, objgrad, obj, cons_coord, cons, hess_coord, hess, hprod, hprod!
 
+"""objcons(nlp, x)
+
+  Computes the objective function and constraint vector values at point x.
+  Internally uses `cfn` or `ufn`.
```
Comment (dpo) [inline src/julia_interface.jl]: Do you think it's necessary to say this in the Julia interface?

File: src/julia_interface.jl [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+  Usage:
+
+      f, c = objcons(nlp, x) # If the problem is constrained
+      f = objcons(nlp, x) # If the problem is unconstrained
```
Comment (dpo) [inline src/julia_interface.jl]: Would you mind aligning the `#`? (Sorry for nitpicking.)

File: src/julia_interface.jl [hunk 3]
```diff
@@ -1,5 +1,19 @@
 export objcons, objgrad, obj, cons_coord, cons, hess_coord, hess, hprod, hprod!
 
+"""objcons(nlp, x)
+
+  Computes the objective function and constraint vector values at point x.
```
Comment (dpo) [inline src/julia_interface.jl]: Just "at x" would be better I believe.

File: src/julia_interface.jl [hunk 4]
```diff
@@ -20,7 +34,22 @@ function objcons(nlp :: CUTEstModel, x :: Array{Float64,1})
   return ncon > 0 ? (f[1], c) : f[1];
 end
 
+"""objgrad(nlp, x, grad)
+
+  Computes the objective function value and, if grad is true, gradient at point
```
Comment (dpo) [inline src/julia_interface.jl]: Should it be `true` so it renders properly in Markdown?

File: src/julia_interface.jl [hunk 5]
```diff
@@ -44,7 +73,24 @@ end
 
 obj(nlp :: CUTEstModel, x :: Array{Float64,1}) = objgrad(nlp, x, false);
 
+"""cons_coord(nlp, x, grad)
 
+  Computes the constraint vector and, if grad is true, the Jacobian in triplet
```
Comment (dpo) [inline src/julia_interface.jl]: Better say "coordinate" format so the function name becomes clear.

File: src/julia_interface.jl [hunk 6]
```diff
@@ -65,6 +111,22 @@ function cons_coord(nlp :: CUTEstModel, x :: Array{Float64,1}, grad :: Bool)
   return grad ? (c, jrow, jcol, jval) : c;
 end
 
+"""cons(nlp, x, grad)
```
Comment (dpo) [inline src/julia_interface.jl]: Should we rename the last argument `jac`?

File: src/julia_interface.jl [hunk 7]
```diff
@@ -118,10 +237,39 @@ function hess(nlp :: CUTEstModel, x :: Array{Float64,1}, y :: Array{Float64,1})
...
 function hess(nlp :: CUTEstModel, x :: Array{Float64,1})
   hess(nlp, x, zeros(nlp.meta.ncon))
 end
 
+"""hprod(nlp, x, y, v)
+
+  Computes the matrix-vector product between the Hessian matrix and the vector
+  v.
+  If the problem is constrained, the Hessian is of the Lagrangian function at
+  point x with Lagrang multipliers y, otherwise the Hessian is of the objective
```
Comment (dpo) [inline src/julia_interface.jl]: Lagrange

File: src/julia_interface.jl [hunk 8]
```diff
@@ -142,6 +290,23 @@ function hprod(nlp :: CUTEstModel, x :: Array{Float64,1}, y :: Array{Float64,1},
   return hv;
 end
 
+"""hprod!(nlp, x, y, v, Hv)
+
+  Computes the matrix-vector product between the Hessian matrix and the vector
+  v and write the result to vector Hv.
+  If the problem is constrained, the Hessian is of the Lagrangian function at
+  point x with Lagrang multipliers y, otherwise the Hessian is of the objective
```
Comment (dpo) [inline src/julia_interface.jl]: Lagrange

**Comment by dpo** [comment]
Many thanks for this! I especially like the description of the input and output arguments!

**Comment by abelsiqueira** [comment]
I think I corrected everything

**Pull request merged**